### PR TITLE
We now look up the name of the tape partition from the Partition Id

### PR DESCRIPTION
**Pull request by scribe**
This is kind of hackey looking for feedback

### CODE REVIEW

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/PhysicalPlacementTapeEntryModel.java [hunk 1]
```diff
@@ -87,8 +95,18 @@ public String getLastModified() {
         return lastModified;
     }
 
-    public String getTapePartition() {
-        return tapePartition;
+    public String getTapePartition() throws IOException {
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/PhysicalPlacementTapeEntryModel.java]: What thread are we doing this in?

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/PhysicalPlacementTapeEntryModel.java [hunk 2]
```diff
@@ -87,8 +95,18 @@ public String getLastModified() {
...
+        try {
+            return ds3Common
+                    .getCurrentSession()
+                    .getClient()
+                    .getTapePartitionSpectraS3(new GetTapePartitionSpectraS3Request(tapePartition))
+                    .getTapePartitionResult()
+                    .getName();
+        } catch (final IOException e) {
+            LOG.error("Unable to get name of tape Partition", e);
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/PhysicalPlacementTapeEntryModel.java]: We should probably log something to the logging service if this happens.

**Comment by scribe** [comment]
I found some code I really don't like in the Physical Placement Task, I'm looking at how to fix it.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/PoolEntry.kt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package com.spectralogic.dsbrowser.gui.components.physicalplacement
```
Comment (RachelTucker) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/PoolEntry.kt]: missing copyright

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/ReplicationEntry.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.spectralogic.dsbrowser.gui.components.physicalplacement
```
Comment (RachelTucker) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/ReplicationEntry.kt]: missing copyright

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/TapeEntry.kt [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.spectralogic.dsbrowser.gui.components.physicalplacement
```
Comment (RachelTucker) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/TapeEntry.kt]: missing copyright

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/PhysicalPlacementModel.kt [hunk 6]
```diff
@@ -13,35 +13,6 @@
...
-        return S3poolType;
-    }
-
-    public String getPartition() {
-        return partition;
-    }
-}
+package com.spectralogic.dsbrowser.gui.components.physicalplacement
 
+data class PhysicalPlacementModel(val poolEntries: List<PoolEntry>, val replicationEntries: List<ReplicationEntry>, val tapeEntries: List<TapeEntry>)
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/PhysicalPlacementModel.kt]: Should these lists be Immutable?

Comment (rpmoore) [reply dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/PhysicalPlacementModel.kt]: Or do they need to be observable collections?

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/PhysicalPlacementTask.java [hunk 7]
```diff
@@ -16,87 +16,177 @@
...
-            this.bucketName = bucketName;
-            this.directoryFullName = directoryFullName;
-            this.ds3Common = ds3Common;
+    private String getPartitionName(final UUID id) {
+        try {
+            return ds3Common.getCurrentSession().getClient()
+                    .getTapePartitionSpectraS3(new GetTapePartitionSpectraS3Request(id.toString()))
+                    .getTapePartitionResult()
+                    .getName();
+        } catch (final IOException e) {
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/PhysicalPlacementTask.java]: This needs to have a LOG.error statement that logs the exception in addition to the loggingService call.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/PhysicalPlacementModel.kt (hunk 6 above)
Comment (scribe) [reply dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/PhysicalPlacementModel.kt]: Aren't kotlin Lists immutable by default?

Comment (rpmoore) [reply dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/PhysicalPlacementModel.kt]: I believe you are correct, but I prefer the explicitness of Guava's ImmutableList.  That and it's more consistent with the rest of the code base.

Comment (scribe) [reply dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/physicalplacement/PhysicalPlacementModel.kt]: :+1:  I'll get that switched over

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/PhysicalPlacementTask.java (hunk 7 above)
Comment (scribe) [reply dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/PhysicalPlacementTask.java]: This should be fixed too

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/ds3Panel/Ds3PanelService.java [hunk 8]
```diff
@@ -172,56 +170,32 @@ public void refresh(final TreeItem<Ds3TreeTableValue> modifiedTreeItem) {
...
-        })));
+        getSelectedItems()
+                .stream()
+                .findFirst()
+                .ifPresent(ds3TreeTableValueTreeItem -> {
+                    final PhysicalPlacementTask getPhysicalPlacement = physicalPlacementTaskFactory.create(ds3TreeTableValueTreeItem.getValue());
+                    workers.execute(getPhysicalPlacement);
+                    getPhysicalPlacement.setOnSucceeded(SafeHandler.logHandle(event -> Platform.runLater(() -> {
+                        LOG.info("Launching PhysicalPlacement popup");
+                        physicalPlacementPopup.show((PhysicalPlacementModel) getPhysicalPlacement.getValue());
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/ds3Panel/Ds3PanelService.java]: Is this cast needed?

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/ds3Panel/Ds3PanelService.java [hunk 9]
```diff
@@ -172,56 +170,32 @@ public void refresh(final TreeItem<Ds3TreeTableValue> modifiedTreeItem) {
...
-        })));
+        getSelectedItems()
+                .stream()
+                .findFirst()
+                .ifPresent(ds3TreeTableValueTreeItem -> {
+                    final MetadataTask getMetadata = new MetadataTask(ds3Common, ImmutableList.of(ds3TreeTableValueTreeItem));
+                    workers.execute(getMetadata);
+                    getMetadata.setOnSucceeded(SafeHandler.logHandle(event -> Platform.runLater(() -> {
+                        LOG.info("Launching metadata popup");
+                        final MetadataView metadataView = new MetadataView((Ds3Metadata) getMetadata.getValue());
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/ds3Panel/Ds3PanelService.java]: Is this cast needed?

**Pull request merged**